t/162-socket-tls-handshake.t

```diff
@@ -0,0 +1,439 @@
+# vim:set ft= ts=4 sw=4 et fdm=marker:
```

Comment:
I am wondering if this new test suite belongs to lua-resty-core instead. After all, all FFI-based APIs are tested there, and not in lua-nginx-module.

Reply:
Yes I thought about this initially, but seems that this clear split is not being followed by existing tests like https://github.com/openresty/lua-nginx-module/blob/master/t/154-semaphore.t.

But, I don't have strong feeling about where to keep the cases, if moving is a better in your opinion then let's move it.